cmd/maya-apiserver/cstor-operator/cspc/storagepool_create.go

```diff
@@ -0,0 +1,362 @@
+/*
+Copyright 2019 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cspc
+
+import (
+	apis "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	apiscsp "github.com/openebs/maya/pkg/cstor/newpool/v1alpha3"
+	container "github.com/openebs/maya/pkg/kubernetes/container/v1alpha1"
+	deploy "github.com/openebs/maya/pkg/kubernetes/deployment/appsv1/v1alpha1"
+	pts "github.com/openebs/maya/pkg/kubernetes/podtemplatespec/v1alpha1"
+	volume "github.com/openebs/maya/pkg/kubernetes/volume/v1alpha1"
+	"github.com/openebs/maya/pkg/version"
+	"github.com/pkg/errors"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"os"
+)
+
+// OpenEBSServiceAccount name of the openebs service accout with required
+// permissions
+const (
+	OpenEBSServiceAccount = "openebs-maya-operator"
+	// PoolMgmtContainerName is the name of cstor target container name
+	PoolMgmtContainerName = "cstor-pool-mgmt"
+
+	// PoolContainerName is the name of cstor target container name
+	PoolContainerName = "cstor-pool"
+
+	// PoolExporterContainerName is the name of cstor target container name
+	PoolExporterContainerName = "maya-exporter"
+)
+
+var (
+	// run container in privileged mode configuration that will be
+	// applied to a container.
+	privileged            = true
+	defaultPoolMgmtMounts = []corev1.VolumeMount{
+		corev1.VolumeMount{
+			Name:      "device",
+			MountPath: "/dev",
+		},
+		corev1.VolumeMount{
+			Name:      "tmp",
+			MountPath: "/tmp",
+		},
+		corev1.VolumeMount{
+			Name:      "udev",
+			MountPath: "/run/udev",
+		},
+	}
+	// hostpathType represents the hostpath type
+	hostpathTypeDirectory = corev1.HostPathDirectory
+
+	// hostpathType represents the hostpath type
+	hostpathTypeDirectoryOrCreate = corev1.HostPathDirectoryOrCreate
+)
+
+// CreateStoragePool creates the required resource to provision a cStor pool
+func (pc *PoolConfig) CreateStoragePool(cspcGot *apis.CStorPoolCluster) error {
+	cspObj, err := pc.AlgorithConfig.GetCSPSpec()
+	if err != nil {
+		return errors.Wrap(err, "failed to get CSP spec")
+	}
+	err = pc.createCSP(cspObj)
+
+	if err != nil {
+		return errors.Wrapf(err, "failed to create csp for cspc {%s}", cspcGot.Name)
+	}
+
+	poolDeployObj := pc.GetPoolDeploySpec(cspObj)
```

Comment:
need to handle the case where deployment fails, but, CSP is created.
This causes issue as the pendingPoolCount during reconcilation checks for CSP count, but not pool deploy count